pages/api/awards/edit.js

```diff
@@ -12,28 +12,63 @@ export default async (req, res) => {
   }
 
   // Collect award information from request body
-  const { id, title, isFinalized } = req.body;
+  const { id, title, settingIDs, isFinalized } = req.body;
 
   if (!id) {
     return res.status(400).json({
       error: 'Required award id not provided',
     });
   }
 
+  // create new settingIDs, delete existing ones
+
   const editData = {};
   if (title) editData.title = title;
   if (isFinalized === false) editData.is_finalized = isFinalized;
 
-  const updatedAward = await prisma.award.update({
+  const deletedAwardCategories = prisma.awardCategory.deleteMany({
+    where: {
+      award_id: id,
+    },
+  });
+
+  let awardCategories = [];
+
+  if (settingIDs && settingIDs.length > 0) {
+    awardCategories = settingIDs.map(settingID =>
+      prisma.awardCategory.upsert({
+        where: {
+          awards_categories_unique: {
+            award_id: id,
+            category_id: settingID,
+          },
+        },
+        create: {
+          award_id: id,
+          category_id: settingID,
+        },
+        update: {},
+      })
+    );
+  }
+
+  const updatedAward = prisma.award.update({
     where: {
       id,
     },
     data: editData,
   });
 
+  const awardTransaction = await prisma.$transaction([
+    deletedAwardCategories,
+    ...awardCategories,
```

Comment:
How come it's `...awardCategories` instead of just `awardCategories`

Reply:
I'm extending the entire array.